ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,383 @@
+        if(_.has(studentObject, 'learningDisabilities') && studentObject.learningDisabilities !== null){ text+= "\n" + studentObject.learningDisabilities };
+        if(_.has(studentObject, 'academicPerformance') && studentObject.academicPerformance !== null){ text+= "\n" + studentObject.academicPerformance };
+        if(_.has(studentObject, 'interests') && studentObject.interests !== null){ text+= "\n" + studentObject.interests };
+        if(_.has(studentObject, 'familyBackground') && studentObject.familyBackground !== null){ text+= "\n" + studentObject.familyBackground };
+        if(_.has(studentObject, 'ses') && studentObject.ses !== null){ text+= "\n" + studentObject.ses };
+        const student = {
+          type: 'student',
+          text: text,
+          key: 'studentInfo'
+        }
+        messages.push(student);
+      }
+    }
+    return messages;
+  },
+  
+  componentDidMount() {
+    setInterval(this.updateTimer, ONE_SECOND);
+  },
+
+  updateTimer() {
+    if(!this.isReadyToMoveOn()){
+      this.setState({ elapsedMs: this.state.elapsedMs + ONE_SECOND });
+    }
+  },
+  
+  componentWillMount(){
+    this.messageId = parseInt(this.props.question.id)*10 + 1;
+  },
+  
+  componentWillReceiveProps(nextProps){
+    var messages = this.getInitialMessages(nextProps);
+    const goodExamples = _.map(nextProps.question.examples, (example) => {
+      return {type: 'Good', text: example};
+    });
+    const badExamples = _.map(nextProps.question.nonExamples, (example) => {
+      return {type: 'Bad', text: example};
+    });
+    const allExamples = _.shuffle(_.clone(goodExamples).concat(badExamples)); 
+    //this.addMessages(messages);
+    const elapsedMs = 0;
+    this.setState({ messages, allExamples, elapsedMs: 0 });
+  },
+  
+  componentDidUpdate(prevProps, prevState){
+    if(prevState.elapsedMs === 90000){
+      this.addMessages([{type: 'info', text: 'Remember, these are supposed to be quick responses.'}]);
+    }
+    if(prevState.initialResponse === undefined && this.state.initialResponse !== undefined){
+      this.logResponse();
+    }else if(prevState.revisedResponse === undefined && this.state.revisedResponse === ''){
+      this.logRevisionDeclined();
+    }else if(prevState.revisedResponse === undefined && this.state.revisedResponse !== undefined){
+      this.logRevision(this.state.revisedResponse);
+    }
+    if(prevState.initialResponse === undefined && this.state.initialResponse !== undefined && this.props.helpType === 'feedback'){
+      this.sendRevision();
+    }
+  },
+  
+  render(){
+    const {hasStarted, onStartPressed, email, query, questionsAnswered, sessionLength} = this.props
+    return (
+      <div>
+        <AppBar
+          style={styles.appBar}
+          title="Message PopUp Mobile"
+          iconElementLeft={
+            <IconButton
+              onTouchTap={this.resetExperience}
+            >
+              {!hasStarted &&
```

Comment:
This seems like it could be more readable as one line since these are so short and wouldn't line-wrap, up to you.